Sources/App/Views/Blog/BlogActions+Model.swift

```diff
@@ -0,0 +1,132 @@
+// Copyright Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import Plot
+import Yams
+import Ink
+
+enum BlogActions {
+
+    struct Model {
+
+        struct PostSummary: Equatable {
+            var slug: String
+            var title: String
+            var summary: String
+            var publishedAt: Date
+            var published: Bool
+        }
+
+        var summaries: [PostSummary]
+
+        init() throws {
+            let blogIndexYmlPath = Current.fileManager.workingDirectory()
+                .appending("Resources/Blog/posts.yml")
+            do {
+                let allSummaries = if let ymlData = Current.fileManager.contents(atPath: blogIndexYmlPath),
+                    let ymlString = String(data: ymlData, encoding: .utf8) {
+                        // Post order should be as exists in the file, but reversed.
+                        try Array(YAMLDecoder().decode([PostSummary].self, from: ymlString).reversed())
+                    } else {
+                        Array<PostSummary>()
+                    }
+
+                summaries = if Current.environment() == .production {
+                    // Only "published" posts show in production.
+                    allSummaries.filter { $0.published }
+                } else {
+                    // Show *everything* in development and on staging.
+                    allSummaries
+                }
+            } catch {
+                Current.logger().report(error: error)
+                summaries = []
+            }
```

Comment:
We should either rethrow the exception or _not_ mark `init` as `throws`. Right now it's `throws` but doesn't need to be.